crates/consensus/src/leader_selection.rs

```diff
@@ -56,7 +56,7 @@ impl WeightedRoundRobin {
     fn proposer_selection(&mut self) -> PeerId {
         let total_voting_power: u64 = self.validators.iter().map(|v| v.voting_power).sum();
 
-        let mut updated_validators = HashSet::new();
+        let mut updated_validators = HashSet::with_capacity(self.validators.len());
```

Comment:
Why not just:

```
self.validators = self.validators.drain().map(|v| { v.priority += v.voting_power as f64 }).collect();
```

You could also define the priority adjusting operations as methods on the AngstromValidator struct so they're a bit more centrally defined

Reply:
you can collect, directly into a HashSet

Reply:
> do you mean
> 
> ```rust
>         self.validators = self.validators.drain().map(|v| { v.priority += v.voting_power as f64; v }).collect::<HashSet<AngstromValidator>>();
> ```
> 
> the same error. you can try it. @Will-Smith11
this will fix it. you are using a map and then mapping the type to () instead of updating power and returning the validaor

Reply:
https://github.com/SorellaLabs/angstrom/pull/247

Reply:
what i said works